client/helpers.go

```diff
@@ -290,3 +283,43 @@ func IsAWSErrors(err error, code []string) bool {
 	}
 	return false
 }
+
+// TagsIntoMap expects []T (usually "[]Tag") where T has "Key" and "Value" fields (of type *string) and writes them into the given map
+func TagsIntoMap(tagSlice interface{}, dst map[string]string) {
+	if k := reflect.TypeOf(tagSlice).Kind(); k != reflect.Slice {
+		panic("invalid usage: Only slices are supported as input: " + k.String())
+	}
+	slc := reflect.ValueOf(tagSlice)
+
+	for i := 0; i < slc.Len(); i++ {
+		val := slc.Index(i)
+		if k := val.Kind(); k != reflect.Struct {
+			panic("slice member is not struct: " + k.String())
+		}
+
+		keyField, valField := val.FieldByName("Key"), val.FieldByName("Value")
+		if keyField.IsZero() || valField.IsZero() {
```

Comment:
not correct, it will panic if the tag is of `Key=nil` and `Value=nil`. Will fix that edge case, but don't think it will happen.

Reply:
Its not obvious that the structs fields Value and Key need to be *string, if you get:

type Tag { Key *string Value string } this won't work and the IsNil will cause a panic. 

I don't know if AWS has such a an option. 

might as well create a better function using https://github.com/mitchellh/mapstructure to convert any struct into a map and return it as tags regardless of the structure, this function be in the SDK basically a convert struct/struct array to map function

Reply:
AWS doesn't have such an option, and if it does we can change the fn. I don't understand how it would work with mapstructure.

I'll change the fn right now.